src/utils.rs

```diff
@@ -9,3 +11,25 @@ pub fn path_to_rel(path: &Path) -> &Path {
         path
     }
 }
+
+#[cfg(feature = "novideo")]
+pub fn read_txt_from_buf<B: AsRef<[u8]>>(buf: B) -> Result<ultrastar_txt::TXTSong> {
+    let reader = buf.as_ref();
+
+    // detect encoding and decode to String
+    let chardet_result = chardet::detect(&reader);
+    let whtwg_label = chardet::charset2encoding(&chardet_result.0);
+    let coder = encoding::label::encoding_from_whatwg_label(whtwg_label);
+    let txt = match coder {
+        Some(c) => match c.decode(&reader, encoding::DecoderTrap::Ignore) {
+            Ok(x) => x,
```

Comment:
While looking at my changes to ultrastar-txt I noticed that this does not work for files encoded with utf8 that sport a BOM at the beginning. 

See here for how I currently implement that check: https://github.com/man0lis/ultrastar-txt/compare/master...e83406f